src/App.tsx

```diff
@@ -1,26 +1,173 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { updateTodo, USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
+import ErrorNotifacations from './components/errorNotifacations';
+import Footer from './components/footer';
+import TodoList from './components/todoList';
+import Header from './components/header';
+import { Todo } from './types/Todo';
+import { filterData } from './helpers/filterData';
+import { ErrorMesagges, FilterOptions } from './types/enums';
+import { deleteTodos } from './api/todos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [errorMessage, setErrorMessage] = useState<ErrorMesagges>(
+    ErrorMesagges.defaultValue,
+  );
+  const [filterTypeValue, setFilterTypeValue] = useState<FilterOptions>(
+    FilterOptions.All,
+  );
+
+  const [deletingIds, setDeletingIds] = useState<number[] | []>([]);
+  const [completedIds, setCompletedIds] = useState<number[] | []>([]);
+
+  const [isSuccessDeleting, setIsSuccessDeleting] = useState<boolean>(false);
+
+  const [isLoadingIds, setIsLoadingIds] = useState<number[] | []>([]);
```

Comment:
```suggestion
  const [deletingIds, setDeletingIds] = useState<number[] | []>([]);
  const [completedIds, setCompletedIds] = useState<number[] | []>([]);

  const [isLoadingIds, setIsLoadingIds] = useState<number[] | []>([]);
```
isSuccessDeleting - we should get rid of this state, can't understand why we need them
3 other states - why we need 3 states they all number[] can we have just one state? like processingIds

Reply:
isSuccessDeleting -- I changed the logic and deleted this state

There three other states exist because they represent different kinds of Ids